pkg/util/workloadrepo/worker_test.go

```diff
@@ -144,3 +169,31 @@ func TestGlobalWorker(t *testing.T) {
 		return len(res) >= 1
 	}, time.Minute, time.Second)
 }
+
+func TestAdminWorkloadRepo(t *testing.T) {
+	ctx, store, dom, addr := setupDomainAndContext(t)
+	tk := testkit.NewTestKit(t, store)
+
+	_, ok := dom.InfoSchema().SchemaByName(model.NewCIStr("workload_schema"))
+	require.False(t, ok)
+
+	wrk := setupWorker(ctx, t, addr, dom, "worker", false)
+	tk.MustExec("set @@global.tidb_workload_repository_snapshot_interval='5000'")
+	tk.MustExec("set @@global.tidb_workload_repository_active_sampling_interval='600'")
+	tk.MustExec("set @@global.tidb_workload_repository_dest='table'")
+
+	require.Eventually(t, func() bool {
+		return wrk.checkTablesExists(ctx)
+	}, time.Minute, time.Second)
+
+	// able to snapshot manually
+	tk.MustExec("admin create workload snapshot")
+	require.Eventually(t, func() bool {
+		res := tk.MustQuery("select snap_id, count(*) from workload_schema.hist_snapshots group by snap_id").Rows()
+		return len(res) >= 1
+	}, time.Minute, time.Second)
+
+	// disable the worker and it will fail
+	tk.MustExec("set @@global.tidb_workload_repository_dest='ble'")
```

Comment:
Is 'ble' a special value or do you just want to set it to an invalid value?

Reply:
> Is 'ble' a special value or do you just want to set it to an invalid value?

Yes

Reply:
Invalid.